src/qmapshack/poi/CPoiDraw.cpp

```diff
@@ -188,6 +188,32 @@ void CPoiDraw::findPoisIn(const QRectF &degRect, QList<poi_t> & pois) const
     CPoiItem::mutexActivePois.unlock();
 }
 
+bool CPoiDraw::getToolTip(const QPoint &px, QString &str)
+{
+    if(CPoiItem::mutexActivePois.tryLock() && poiList)
```

Comment:
:rotating_light: Potential dead lock. Same as in `findPOICloseBy`. If the mutex locks but there is no `poiList` we will leave without unlocking the mutex. 

I just ran into the situation that the thread to retrieve and render all POIs is blocking for ever.

Reply:
So there have to be two nested `if` statements, right? Because unlocking when you haven't locked it isn't good either.

Reply:
hm, but even 
```
if(poiList && CPoiItem::mutexActivePois.tryLock())
```
blocks. The must be more deadlocks. Please crosscheck all your mutex locks!

Reply:
Are they fixed now?

Reply:
seems to be ok for now.